host/xtest/pkcs11_1000.c

```diff
@@ -2565,3 +2565,237 @@ static void xtest_pkcs11_test_1011(ADBG_Case_t *c)
+	rv = create_data_object(ro_session, &obj_hdl, CK_TRUE,
+				CK_TRUE, label);
+	if (!ADBG_EXPECT_CK_RESULT(c, CKR_SESSION_READ_ONLY, rv))
+		goto out;
+
+	Do_ADBG_EndSubCase(c, NULL);
+
+	/*
+	 * Sub test: Check access for a R/W User session
+	 */
+	Do_ADBG_BeginSubCase(c, "Create objects in R/W User Session");
+
+	/* Session Public Obj CKA_TOKEN = CK_FALSE, CKA_PRIVATE = CK_FALSE */
+	rv = create_data_object(session, &obj_hdl, CK_FALSE,
+				CK_FALSE, label);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	/* Session Private Obj CKA_TOKEN = CK_FALSE, CKA_PRIVATE = CK_TRUE */
+	rv = create_data_object(session, &obj_hdl, CK_FALSE,
+				CK_TRUE, label);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	/* Token Public Obj CKA_TOKEN = CK_TRUE, CKA_PRIVATE = CK_FALSE */
+	rv = create_data_object(session, &obj_hdl, CK_TRUE,
+				CK_FALSE, label);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	/* Token Private Obj CKA_TOKEN = CK_TRUE, CKA_PRIVATE = CK_TRUE */
+	rv = create_data_object(session, &obj_hdl, CK_TRUE,
+				CK_TRUE, label);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	Do_ADBG_EndSubCase(c, NULL);
+
+	/* Log out */
+	ADBG_EXPECT_CK_OK(c, C_Logout(ro_session));
+	logged_in = false;
+
+	/* Close RO session */
+	ADBG_EXPECT_CK_OK(c, C_CloseSession(ro_session));
+	ro_exists = false;
+
+	/*
+	 * Sub test: Check access for a R/W SO session
+	 */
+	Do_ADBG_BeginSubCase(c, "Create objects in R/W SO Session");
+
+	/* Login as security officer in RW session */
+	rv = C_Login(session, CKU_SO, test_token_so_pin,
+		     sizeof(test_token_so_pin));
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto out;
+
+	/* Session Public Obj CKA_TOKEN = CK_FALSE, CKA_PRIVATE = CK_FALSE */
+	rv = create_data_object(session, &obj_hdl, CK_FALSE,
+				CK_FALSE, label);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto logout;
+
+	/* Session Private Obj CKA_TOKEN = CK_FALSE, CKA_PRIVATE = CK_TRUE */
+	rv = create_data_object(session, &obj_hdl, CK_FALSE,
+				CK_TRUE, label);
+	if (!ADBG_EXPECT_CK_RESULT(c, CKR_USER_NOT_LOGGED_IN, rv))
+		goto logout;
+
+	/* Token Public Obj CKA_TOKEN = CK_TRUE, CKA_PRIVATE = CK_FALSE */
+	rv = create_data_object(session, &obj_hdl, CK_TRUE,
+				CK_FALSE, label);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto logout;
+
+	/* Token Private Obj CKA_TOKEN = CK_TRUE, CKA_PRIVATE = CK_TRUE */
+	rv = create_data_object(session, &obj_hdl, CK_TRUE,
+				CK_TRUE, label);
+	if (!ADBG_EXPECT_CK_RESULT(c, CKR_USER_NOT_LOGGED_IN, rv))
+		goto logout;
+
+	Do_ADBG_EndSubCase(c, NULL);
+
+logout:
+	ADBG_EXPECT_CK_OK(c, C_Logout(session));
+out:
+	if (logged_in)
+		ADBG_EXPECT_CK_OK(c, C_Logout(ro_session));
+
+	if (ro_exists)
+		ADBG_EXPECT_CK_OK(c, C_CloseSession(ro_session));
+
+	ADBG_EXPECT_CK_OK(c, C_CloseSession(session));
+
+	destroy_persistent_objects(c, slot);
+close_lib:
+	ADBG_EXPECT_CK_OK(c, close_lib());
+
+}
+ADBG_CASE_DEFINE(pkcs11, 1012, xtest_pkcs11_test_1012,
+		 "PKCS11: Check access of Objects");
```

Comment:
I would change the description to `"PKCS11: objects creation upon session type"`.

Reply:
Changed